pkg/ingress/model_builder.go

```diff
@@ -276,3 +283,36 @@ func (t *defaultModelBuildTask) mergeListenPortConfigs(_ context.Context, listen
 		tlsCerts:       mergedTLSCerts.List(),
 	}, nil
 }
+
+// buildSSLRedirectConfig computes the SSLRedirect config for the IngressGroup. Returns nil if there is no SSLRedirect configured.
+func (t *defaultModelBuildTask) buildSSLRedirectConfig(ctx context.Context, listenPortConfigByPort map[int64]listenPortConfig) (*SSLRedirectConfig, error) {
+	explicitSSLRedirectPorts := sets.Int64{}
+	for _, ing := range t.ingGroup.Members {
+		var rawSSLRedirectPort int64
+		exists, err := t.annotationParser.ParseInt64Annotation(annotations.IngressSuffixSSLRedirect, &rawSSLRedirectPort, ing.Annotations)
+		if err != nil {
+			return nil, errors.Wrapf(err, "ingress: %v", k8s.NamespacedName(ing))
+		}
+		if exists {
+			explicitSSLRedirectPorts.Insert(rawSSLRedirectPort)
+		}
+	}
+
+	if len(explicitSSLRedirectPorts) == 0 {
+		return nil, nil
+	}
+	if len(explicitSSLRedirectPorts) > 1 {
+		return nil, errors.Errorf("conflicting sslRedirect port: %v", explicitSSLRedirectPorts.List())
+	}
+	rawSSLRedirectPort, _ := explicitSSLRedirectPorts.PopAny()
+	if listenPortConfig, ok := listenPortConfigByPort[rawSSLRedirectPort]; !ok {
+		return nil, errors.Errorf("listener non-exists for SSLRedirect port: %v", rawSSLRedirectPort)
```

Comment:
"listener non-existent ... " or "listener does not exist ... " looks better to me.

Reply:
fixed :D make sense